src/mono/mono/sgen/sgen-scan-object.h

```diff
@@ -42,6 +42,10 @@
 #if defined(SGEN_HEAVY_BINARY_PROTOCOL) && defined(SCAN_OBJECT_PROTOCOL)
 	sgen_binary_protocol_scan_vtype_begin (start + SGEN_CLIENT_OBJECT_HEADER_SIZE, size);
 #endif
+#endif
+#if defined(ENABLE_NETCORE) && !defined(DISABLE_ALC_UNLOADABILITY)
+	// Scan RefTracker for collectible MemoryManagers
+	// This is expensive, and should probably be optimized by putting the RefTracker handle in the vtables when appropriate so we can just check there
```

Comment:
It's supposed to scan the `obj->vtable->klass->image->alc->ref_tracker` gc handle for a non-generic object and
for a generic instance it's `each_ref_tracker_of_image_set (((MonoClassGenericInst*)obj->vtable->klass)->generic_class->owner)` something like that?

So if we put something in the vtable directly, for a generic instance we'd need some kind of `GenericRefTracker` that points at an array of regular ref trackers, right?